ReviewChecklist.md

```diff
@@ -0,0 +1,156 @@
+- [ ] **Mark hacks as such.** If you have to leave behind something ugly, make sure to include a
+    comment that explains why and in what circumstances it can be removed.
+    Preferably link to an issue you reported upstream.
+- [ ] **Avoid obvious comments.**
+- [ ] **Do include comments when the reader may need extra context to understand the code.**
+
+### Commits and PRs
+- [ ] **Avoid hiding functional changes inside refactors.**
+    E.g. when fixing a small bug, or changing user-visible behavior, put the change in a separate commit.
+    Do not mix it with another change that renames things or reformats the code around, making the fix itself hard to identify.
+- [ ] **Whenever possible, split off refactors or unrelated changes into separate PRs.**
+    Smaller PRs are easier and quicker to review.
+    Splitting off refactors helps focus on the main point of the PR.
+
+### Common pitfalls
+These are covered by the coding style but come up so often that it's worth singling them out here:
+- [ ] **Always initialize value types in the definition,** even if you're sure you will assign them later.
+- [ ] **Use "east const" style.** I.e. `T const*`, not `const T *`.
+- [ ] **Keep indentation consistent.** See our [`.editorconfig`](.editorconfig).
+    - [ ] Tabs for C++. But use them for indentation only. Any whitespace later on the line must be spaces.
+    - [ ] 4 spaces for most other file types.
+- [ ] **Use `auto` sparingly.** Only when the actual type is very long and complicated or when it's
+    already used elsewhere in the same expression.
+- [ ] **Indent braces and parentheses in a way that makes nesting clear.**
+- [ ] **Use `using namespace` only in `.cpp` files.** Use it for `std` and our own modules.
+    Avoid unnecessary `std::` prefix in `.cpp` files (except for very common names like `std::move`).
+- [ ] **Use range-based loops and destructuring.**
+- [ ] **Include any headers you use directly,** even if they're implicitly included through other headers.
+
+## Documentation
+- [ ] **Does the PR update relevant documentation?**
+
+### Documentation style and good practices
+- [ ] **Use double backticks in RST (``` ``x`` ```). Prefer single backticks in Markdown (`` `x` ``),**
+    but note that double backticks are valid too and we use them in some cases for legacy reasons.
+- [ ] **Always start a new sentence on a new line.**
+    This way you don't have to rewrap the surrounding text when you rewrite the sentence.
+    This also makes parts that were actually easier to spot in the diff.
+
+## Testing
+
+### What to test
+- [ ] **Is newly added code adequately covered by tests?**
+    - [ ] **Is the (cleaned up) repro from the issue included as a test case?**
+    - [ ] If it's a bugfix: **Do tests cover not just the bug repro but also related cases that might be affected by a similar bug?**
+- [ ] **Positive cases should have a semantic test.**
+- [ ] **Negative cases should have a syntax test.**
```

Comment:
'''
- [ ] **Positive cases ( i.e. code behaving as per expectations ) should have a semantic test.**
- [ ] **Negative cases (i.e. code that results in errors ) should have a syntax test.**
```